components/AnimatedLoading.js

```diff
@@ -0,0 +1,85 @@
+import React, { useCallback, useEffect, useRef } from 'react';
+import { Animated } from 'react-native';
+import LottieView from 'lottie-react-native';
+import PropTypes from 'prop-types';
+import { connectStyle } from '@shoutem/theme';
+import { View } from '@shoutem/ui';
+import { animations } from '../assets';
+import { usePreviousValue } from '../hooks';
+
+const AnimatedView = Animated.createAnimatedComponent(View);
+
+function AnimatedLoading({
```

Comment:
allow sending the animation to this component, with the default one being the one you're using right now